docs/source/dev/how-io-works.rst

```diff
@@ -0,0 +1,134 @@
+  implementing a :code:`Decoder`, or even completely overriding
+  the :code:`Partition.get_tiles` functionality.
+- :code:`IOBackend.get_tiles` has two modes of operation: either it reads the
+  tiles "straight" from the file, without copying or decoding, or it
+  uses the read ranges and copies/decodes the tiles in smaller units.
+- When reading the tiles "straight", the read ranges are not used, instead
+  only the slice information for each tile is used. That also means that this
+  mode only works for very simple formats, when reading without a :code:`roi`
+  and when not doing any :code:`dtype` conversion or decoding.
+
+Read ranges
+-----------
+
+In :code:`FileSet.get_read_ranges`, the reading parameters (TilingScheme, roi etc.)
+are translated into one or more byte ranges (offset, length) for each tile.
+You can imagine it as translating pixel/element positions into byte offsets.
+
+Each range corresponds to a read operation on a single file, which means read
+ranges for a single tile can correspond to reads from multiple files. This
+is important when reading from a data set with many small files - we can
+still generate tiles for efficient processing.
+
+There are some built-in common parameters in `FileSet`, like
+`frame_header_bytes`, `frame_footer_bytes`, which can be used to easily
+implement formats where the reading just needs to skip a few bytes for each
+frame header/footer.
+
+If you need more influence over how data is read, you can override
+`FileSet.get_read_ranges` and return your own read ranges. You can use
+the `make_get_read_ranges` function to re-use much of the tiling logic,
+or implement this yourself. Using `make_get_read_ranges` you can either
+override just the `px_to_bytes` part, or `read_ranges_tile_block` for whole
+tile blocks. This is done by passing in njit-ed functions to `make_get_read_ranges`.
+`make_get_read_ranges` should only be called on module-level to enable
+caching of the numba compilation.
+
+Read ranges are generated as an array with the following shape::
+
+    (number_of_tiles, rr_per_tile, rr_num_entries)
+
+:code:`rr_per_tile` here is the maximum number of read ranges per tile - there
+can be tiles that are smaller than this, for example at the end of a partition.
+:code:`rr_num_entries` is at least 3 and contains at least the values
+:code:`(file_idx, start, stop)`. This means to read :code:`stop - start`
+bytes, beginning at offset :code:`start`, from the file :code:`file_idx`
+in the corresponding :code:`FileSet`.
+
+Overriding :code:`DataSet`\ s are free to add additional fields to the end, for
+example if the decoding functions need additional information.
+
+As an example when you would generate custom read ranges, have a look at the
+implementations for MIB, K2IS, and FRMS6 - they may not have a direct 1:1 mapping
+to a numpy :code:`dtype`, or the pixels may need to be re-ordered after decoding.
+In these cases, 
```

Comment:
Incomplete sentence: "In these cases, ..."

Reply:
Good catch! Removing for now, can add more detail later.